### DISCUSSION TITLE
Is it possible to change colours via i3-msg?

### DISCUSSION THREAD

**Discussion by DavidAntliff** [i3/i3#6166] [Q&A] [answered]
I'm using i3 4.23, bash 5.1.16.

I wish to change some i3 colours from the shell, so I have been trying the following, based on the default values from [client_colors](https://i3wm.org/docs/userguide.html#client_colors):

```
$ i3-msg "client.unfocused #333333 #222222 #888888 #292d2e #222222"
ERROR: Your command: client.unfocused #333333 #222222 #888888 #292d2e #222222
ERROR:               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR: Expected one of these tokens: <end>, '[', 'move', 'exec', 'exit', 'restart', 'reload', 'shmlog', 'debuglog', 'border', 'layout', 'append_layout', 'workspace', 'focus', 'kill', 'open', 'fullscreen', 'sticky', 'split', 'floating', 'mark', 'unmark', 'resize', 'rename', 'nop', 'scratchpad', 'swap', 'title_format', 'title_window_icon', 'mode', 'bar', 'gaps'
[{"success":false,"parse_error":true,"error":"Expected one of these tokens: <end>, '[', 'move', 'exec', 'exit', 'restart', 'reload', 'shmlog', 'debuglog', 'border', 'layout', 'append_layout', 'workspace', 'focus', 'kill', 'open', 'fullscreen', 'sticky', 'split', 'floating', 'mark', 'unmark', 'resize', 'rename', 'nop', 'scratchpad', 'swap', 'title_format', 'title_window_icon', 'mode', 'bar', 'gaps'","input":"client.unfocused #333333 #222222 #888888 #292d2e #222222","errorposition":"^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^"}]
```

I've also tried single-quotes instead of double quotes, and '\#' and '\\#' but I get the same error in every case.

It seems that `client.*` isn't listed as an accepted token - is it not possible to change colours in this way?

**Comment by DavidAntliff** [comment]
I should mention that I also tried:

```
$ i3-msg client.unfocused #333333 #222222 #888888 #292d2e #222222
```
But of course `bash` treats `#` as the start of a comment so that is effectively the same as writing only:

```
$ i3-msg client.unfocused
```

, which is incomplete.

**Reply by okraits** [reply, marked as answer]
No, this is not possible. `client.*` are not commands but config directives. For a list of commands see https://i3wm.org/docs/userguide.html#list_of_commands

To dynamically change i3 colors, you could include another file in your i3 config and then manipulate that other file with shell commands and reload the i3 config afterwards.
